tests/mock/clitools.rs

```diff
@@ -1149,21 +1149,27 @@ where
     A: AsRef<Path>,
     B: AsRef<Path>,
 {
-    drop(fs::remove_file(b.as_ref()));
-    fs::hard_link(a, b).map(|_| ())
+    fn inner(a: &Path, b: &Path) -> io::Result<()> {
+        match fs::remove_file(b) {
+            Err(e) if e.kind() != io::ErrorKind::NotFound => return Err(e),
+            _ => {}
+        }
+        fs::hard_link(a, b).map(drop)
+    }
+    inner(a.as_ref(), b.as_ref())
 }
 
 pub fn copy_binary<A, B>(a: A, b: B) -> io::Result<()>
 where
     A: AsRef<Path>,
     B: AsRef<Path>,
 {
-    eprintln!(
-        "Copying from {} to {}",
-        a.as_ref().display(),
-        b.as_ref().display()
-    );
-    let _lock = cmd_lock().write().unwrap();
```

Comment:
You have lost the protection of the `cmd_lock()` here which will increase the chances of race conditions hitting in the test suite on CI.

Reply:
Done.